File: Python3/108. Convert Sorted Array to Binary Search Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def sortedArrayToBST(self, nums):
        """
        :type nums: List[int]
        :rtype: TreeNode
        """
        if len(nums) == 0: # 72ms
            return None
        elif len(nums) == 1:
            return TreeNode(nums[0])
        else:
            node = TreeNode(nums[len(nums)//2])
            node.left = self.sortedArrayToBST(nums[:len(nums)//2])
            node.right = self.sortedArrayToBST(nums[len(nums)//2+1:])
            return node

File: Golang/108. Convert Sorted Array to Binary Search Tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func sortedArrayToBST(nums []int) *TreeNode { // 264ms
    length := len(nums)
    if length == 0 {
        return nil
    }
    median := length / 2
    node := &TreeNode{}
    node.Val = nums[median]
    node.Left = sortedArrayToBST(nums[: median])
    node.Right = sortedArrayToBST(nums[median+1: ])
    return node
}
